src/main/scala/beam/agentsim/agents/choice/mode/PtFares.scala

```diff
@@ -4,31 +4,74 @@ import java.io.FileNotFoundException
 import java.nio.file.{Files, Paths}
 
 import beam.agentsim.agents.choice.mode.PtFares.FareRule
+import org.slf4j.LoggerFactory
 
 import scala.collection.mutable.ListBuffer
 import scala.io.Source
 import scala.util.Try
 
-case class PtFares(ptFares: Map[String, List[FareRule]]) {
+case class PtFares(ptFares: List[FareRule]) {
 
-  def getPtFare(agencyId: String, routeId: Option[String], age: Option[Int]): Option[Double] = {
-    ptFares
-      .getOrElse(agencyId, List())
+  def getPtFare(agencyId: Option[String], routeId: Option[String], age: Option[Int]): Option[Double] = {
+    val ageFilter = {
+      val ageMatch = ptFares
+        .filter(
+          f =>
+            age.fold(false)(f.age.has)
+        )
+
+      lazy val ageNoMatch = ptFares
```

Comment:
It took a bit of thinking to discern this. However I'm still not sure the naming and the end result match up to me. What is this supposed to yield? I'm wondering if a diff using `ageMatch` would be more easily understood to a reader of the code? Or I missed something..

Reply:
Admittedly, the whole flow takes a bit to understand...Could this be extracted into a method that has a name of what it is doing ... as a guide for those reading the code?